tooling/lsp/src/lib.rs

```diff
@@ -95,6 +95,9 @@ pub struct LspState {
     cached_parsed_files: HashMap<PathBuf, (usize, (ParsedModule, Vec<ParserError>))>,
     cached_def_maps: HashMap<String, BTreeMap<CrateId, CrateDefMap>>,
     options: LspInitializationOptions,
+
+    // Tracks files that currently have errors, by package root.
+    files_with_errors: HashMap<String, HashSet<Url>>,
```

Comment:
Looks like the roots are originally `PathBuf`s, why not keep them as PathBufs instead of going to as_os_str -> to_string_lossy?

Reply:
Good idea, done! I also made that changed for a couple more HashMaps.